include/sempr/processing/CalculatePointsModule.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef SEMPR_PROCESSING_CALCULATEPOINTSMODULE_HPP
+#define SEMPR_PROCESSING_CALCULATEPOINTSMODULE_HPP
+
+#include <geos/geom/Polygon.h>
+#include <geos/geom/Coordinate.h>
+
+#include <sempr/processing/Module.hpp>
+#include <sempr/query/ObjectQuery.hpp>
+#include <sempr/query/CalculatePointsQuery.hpp>
+#include <sempr/query/SpatialIndexQuery.hpp>
+#include <PointCloud_odb.h>
+#include <AgriEntity_odb.h>
+
+namespace sempr { namespace processing {
+
+class CalculatePointsModule : public Module< query::CalculatePointsQuery >
+{
+public:
+    using Ptr = std::shared_ptr<CalculatePointsModule>;
+
+    CalculatePointsModule();
+
+    ~CalculatePointsModule();
+
+    virtual std::string type() const override;
+
+    /* Get the internal Points of an Polygon. Points can be from MultiPoint or PointCloud */
+    void process(query::CalculatePointsQuery::Ptr query) override;
+    
+
+private:
+    int checkIntersection(geom::Coordinate v11, geom::Coordinate v12, geom::Coordinate v21, geom::Coordinate v22);
+
+    void calculatePoints(const entity::PointCloud::Ptr cloud, query::CalculatePointsQuery::Ptr query);
+    
+    void calculatePoints(const entity::PointCloud::Ptr cloud, std::vector <geos::geom::Coordinate> coords);
+    
+    bool intersect(std::vector<geom::Coordinate> c, std::vector <geom::Coordinate> p);
+
+    std::vector<geom::Coordinate> m_c;
```

Comment:
This is basically one pointcloud. Why not using PointCloud and what if there is more than one?

Reply:
There will not be more then one.
`One` `Polygon` gets *one* set of points.
If there would be more [PointClouds / vectors], there would be more polygons **or** intersecting edges, but a `CalculatePointQuery` only gets one `Polygon` and such an `Polygon` is not allowed to have intersecting edges [see here](http://geos.refractions.net/ro/doxygen_docs/html/classgeos_1_1geom_1_1LinearRing.html).